src/plots/Volume/QvisVolumePlotWindow.C

```diff
@@ -1382,19 +1415,60 @@ void QvisVolumePlotWindow::CreateOSPRayGroups(QWidget *parent, QLayout *pLayout)
     mincontributionLayout->addWidget(osprayMinContributionLabel);
     mincontributionLayout->addWidget(osprayMinContribution, Qt::AlignLeft);
     mincontributionLayout->addStretch(QSizePolicy::Maximum);
+
+    // value: max contribution
+    osprayMaxContributionWidget        = new QWidget(osprayGroup);
+    QHBoxLayout *maxcontributionLayout = new QHBoxLayout(osprayMaxContributionWidget);
+    osprayMaxContributionLabel         = new QLabel(tr("Maximum Contribution"),
+                                                    osprayMaxContributionWidget);
+    osprayMaxContribution              = new QDoubleSpinBox(osprayMaxContributionWidget);
+    osprayMaxContribution->setKeyboardTracking(false);
+    osprayMaxContribution->setMaximum(0.001);
+    osprayMaxContribution->setMaximum(10000);
+    osprayMaxContribution->setSingleStep(0.001);
+    osprayMaxContribution->setValue(0.001);
+    osprayMaxContribution->setDecimals(3);
+    osprayMaxContributionLabel->setBuddy(osprayMaxContribution);
+    connect(osprayMaxContribution, SIGNAL(valueChanged(double)),
+            this, SLOT(osprayMaxContributionChanged(double)));
+    maxcontributionLayout->addWidget(osprayMaxContributionLabel);
+    maxcontributionLayout->addWidget(osprayMaxContribution, Qt::AlignLeft);
+    maxcontributionLayout->addStretch(QSizePolicy::Maximum);
+
     // layout
-    osprayGroupLayout->addWidget(osprayShadowToggle,      0,0);
-    osprayGroupLayout->addWidget(ospraySingleShadeToggle, 1,0);
-    osprayGroupLayout->addWidget(osprayUseGridAcceleratorToggle, 0,2);
-    osprayGroupLayout->addWidget(osprayOneSidedLightingToggle,   1,2);
-    osprayGroupLayout->addWidget(osprayPreIntegrationToggle,     0,5,1,5);
-    osprayGroupLayout->addWidget(osprayAoTransparencyToggle,     1,5,1,5);
-    osprayGroupLayout->addWidget(ospraySppWidget,             2,1,1,2,Qt::AlignRight);
-    osprayGroupLayout->addWidget(osprayAoSamplesWidget,       3,1,1,2,Qt::AlignRight);
-    osprayGroupLayout->addWidget(osprayMinContributionWidget, 2,4,1,4,Qt::AlignRight);
-    osprayGroupLayout->addWidget(osprayAoDistanceWidget,      3,4,1,4,Qt::AlignRight);
+    osprayLayout->addWidget( osprayToggle );
+    osprayLayout->addWidget( osprayProperties );
+
+    // osprayPropertiesLayout->addWidget(osprayShadowToggle,             0, 0);
```

Comment:
Are these commented-out settings reserved for future use? If so please add a comment about under what circumstances they should be enabled, otherwise please delete them.

Reply:
Not everything is enabled in OSPRay via VTK - comment added.